tests/cases/fourslash/codeFixClassImplementInterfaceAutoImports.ts

```diff
@@ -28,11 +28,10 @@ verify.codeFix({
   description: "Implement interface 'Base'",
   newFileContent:
 `import { Base } from './interface';
-import A from './types1';
 import { B, C, D } from './types2';
 
 export class C implements Base {
-    a: Readonly<A> & { kind: 'a'; };
```

Comment:
This is a somewhat unfortunate change. However, it matches the behavior of plain `A` that is `{}`. In such a case, `A` doesn't appear in the output after the refactoring. Maybe that's an issue worth raising? The intention of the user could have been to reuse types from the interface quite literally - so when they start implementing `A`'s members the `C` class automatically "receives" them.

Reply:
I agree that it might be unfortunate, but I think this change is ok. If we want to have a mechanism for preserving those kinds of intersections in refactorings and codefixes, we need it to also work for `A & { kind: ... }`.